app/test-utils/testFunctions/questionTests/errorTests.js

```diff
@@ -0,0 +1,154 @@
+      });
+
+    const errorSummary = Selector('[data-test-id="error-summary"]');
+    const errorSummaryList = Selector('.nhsuk-error-summary__list');
+    const renderedQuestion = Selector(`[data-test-id="question-${questionId}"]`);
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+    const expectedErrorMessage = sectionManifest.questions[questionId].errorResponse.maxLength;
+
+    await t
+      .expect(errorSummary.exists).notOk()
+      .typeText(renderedQuestion, `${'x'.repeat(501)}`, { paste: true })
+      .click(submitButton.find('button'))
+      .expect(errorSummary.exists).ok()
+      .expect(errorSummaryList.find('li').count).eql(1)
+      .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedErrorMessage)
+      .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`#${questionType === 'bulletpoint-list' ? `${questionId}-1` : questionId}`)
+      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedErrorMessage}`);
+    if (questionType === 'textarea-field') {
+      await t
+        .expect(renderedQuestion.find('.nhsuk-textarea--error').exists).ok();
+    }
+  });
+};
+
+const mandatoryErrorTest = ({
+  pageSetup,
+  sectionManifest,
+  questionId,
+  sectionApiUrl,
+  apiLocalhost,
+}) => {
+  test(`should show error summary and validation for ${questionId} question indicating it is mandatory`, async (t) => {
+    await pageSetup({ t });
+    nock(apiLocalhost)
+      .put(sectionApiUrl)
+      .reply(400, {
+        required: [questionId],
+      });
+
+    const expectedQuestion = sectionManifest.questions[questionId];
+    const expectedErrorMessage = expectedQuestion.errorResponse.required;
+    const errorSummary = Selector('[data-test-id="error-summary"]');
+    const errorSummaryList = Selector('.nhsuk-error-summary__list');
+    const renderedQuestion = Selector(`[data-test-id="question-${questionId}"]`);
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+    await t
+      .expect(errorSummary.exists).notOk()
+      .click(submitButton.find('button'))
+      .expect(errorSummary.exists).ok()
+      .expect(errorSummaryList.find('li').count).eql(1)
+      .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedErrorMessage)
+      .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`#${questionId}`)
+      .expect(renderedQuestion.find(`[data-test-id="${expectedQuestion.type}-error"]`).exists).ok()
+      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedErrorMessage}`);
+  });
+};
+
+export const runErrorTests = ({
+  pageSetup,
+  sectionManifest,
+  questionId,
+  sectionApiUrl,
+  questionData,
+  sectionId,
+  apiLocalhost,
+  sectionParent,
+}) => {
+  Object.keys(questionData.errorResponse).forEach((errorType) => {
+    if (errorType === 'required') {
+      mandatoryErrorTest({
```

Comment:
Possible to move the guard inside the test function?

Reply:
👍